frappe/www/app.html

```diff
@@ -38,17 +38,16 @@
 		<div id="build-events-overlay"></div>
 
 		<script type="text/javascript">
-		window._version_number = "{{ build_version }}";
-		// browser support
-		window.app = true;
-		window.dev_server = {{ dev_server }};
+			window._version_number = "{{ build_version }}";
+			// browser support
+			window.app = true;
+			window.dev_server = {{ dev_server }};
 
-		if(!window.frappe) window.frappe = {};
-
-		frappe.boot = {{ boot }};
-
-		frappe.csrf_token = "{{ csrf_token }}";
+			if (!window.frappe) window.frappe = {};
 
+			frappe.boot = {{ boot }};
+			frappe._messages = Object.assign({}, frappe.boot["__messages"]);
```

Comment:
Maybe this doesn't need to be a copy at all? 5000 translations can take up quite some memory from the limited memory that a browser allocates to a tab. My understanding is that `frappe.boot.__messages` is used for the sole purpose of updating `frappe._messages`, so simply pointing `frappe._messages` to this object should be safe.